### PR TITLE
[feature] Add necessary checks  

### PR DESCRIPTION
**Pull request by ls-2018**
- Fixed inconsistencies in highlight counts
- The number of titles is inconsistent
- The number of links is inconsistent
- Check the word for correctness
- Check struct demo 
- ci add typo、struct check

### CODE REVIEW

**Comment by ls-2018** [comment]
@furykerry

File: .github/workflows/pre_dict.json [hunk 1]
```diff
@@ -0,0 +1,799 @@
+[
```
Comment (furykerry) [inline .github/workflows/pre_dict.json:1]: plz give some guidelines in README.md about adding new words into pre_dict.json

**Comment by ls-2018** [comment]
@furykerry all issues fixed

**Review by furykerry — COMMENTED** [review]
/lgtm

**Pull request merged**